### PR TITLE
[20509] Update repository for Fast DDS 3.0.0 compatibility

### PR DESCRIPTION
**Pull request by LuciaEchevarria99**
Merge after:
-  **Fast-DDS:** https://github.com/eProsima/Fast-DDS/pull/4779
- **dev-utils:** https://github.com/eProsima/dev-utils/pull/101
- **DDS-Pipe:** https://github.com/eProsima/DDS-Pipe/pull/111
- **DDS-Record-Replay:** https://github.com/eProsima/Fast-DDS-spy/pull/93

### CODE REVIEW

File: fastddsspy_participants/include/fastddsspy_participants/model/NetworkDatabase.hpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 #pragma once
 
-#include <fastrtps/types/DynamicTypePtr.h>
+#include <fastdds/dds/xtypes/dynamic_types/DynamicType.hpp>
```
Comment (juanlofer-eprosima) [inline fastddsspy_participants/include/fastddsspy_participants/model/NetworkDatabase.hpp:17]:
```suggestion
```

File: fastddsspy_participants/test/unittest/model/DataStreamerTest.cpp [hunk 2]
```diff
@@ -15,24 +15,35 @@
...
-        fastrtps::types::DynamicType_ptr& dynamic_type_topic)
+fastdds::dds::DynamicType::_ref_type create_schema(
+        ddspipe::core::types::DdsTopic& topic)
 {
-    fastrtps::types::DynamicTypeBuilder_ptr dynamic_type_topic_builder;
-    dynamic_type_topic_builder = fastrtps::types::DynamicTypeBuilderFactory::get_instance()->create_struct_builder();
-
-    dynamic_type_topic_builder->set_name(topic.type_name);
-    dynamic_type_topic = dynamic_type_topic_builder->build();
+    // // Opción 1:
```
Comment (juanlofer-eprosima) [inline fastddsspy_participants/test/unittest/model/DataStreamerTest.cpp:30]: Please tidy up.

File: fastddsspy_tool/src/cpp/tool/Controller.cpp [hunk 3]
```diff
@@ -141,13 +141,16 @@ void Controller::data_stream_callback_(
     // TODO fast this does not make much sense as dynamictypes::print does not allow to choose target
     // change in dyn types to be able to print it in view
     view_.show("---");
-    fastrtps::types::DynamicDataHelper::print(dyn_data.get());
+    std::stringstream ss;
+    ss << std::setw(4);
+    auto ret = fastdds::dds::json_serialize(dyn_data, fastdds::dds::DynamicDataJsonFormat::EPROSIMA, ss);
```
Comment (juanlofer-eprosima) [inline fastddsspy_tool/src/cpp/tool/Controller.cpp:146]: Print error if not OK?

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: fastddsspy_tool/src/cpp/tool/Controller.cpp (hunk 3 above)
Comment (LuciaEchevarria99) [reply fastddsspy_tool/src/cpp/tool/Controller.cpp:146]: I decided to put a warning

File: fastddsspy_tool/test/labels/XTSAN.list [hunk 4]
```diff
@@ -0,0 +1,66 @@
+<<<<<<< Updated upstream
```
Comment (juanlofer-eprosima) [inline fastddsspy_tool/test/labels/XTSAN.list:1]: Fix this.

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**